odoo/fields.py

```diff
@@ -1448,7 +1448,7 @@ def write(self, records, value):
             for rid in real_recs._ids:
                 # cache_value is already in database format
                 towrite[rid][self.name] = cache_value
-            if self.translate is True and cache_value is not None:
+            if self.translate is True and not cache_value:
```

Comment:
```suggestion
            if self.translate is True and cache_value:
```

right?

Reply:
Thanks. I was about to check why other tests don't work :smiley: